extensions/interactions/NumericExpressionInput/directives/oppia-response-numeric-expression-input.component.ts

```diff
@@ -20,16 +20,29 @@
  * followed by the name of the arg.
  */
 
-require('services/html-escaper.service.ts');
+import { Component, Input, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { HtmlEscaperService } from 'services/html-escaper.service';
 
-angular.module('oppia').component('oppiaResponseNumericExpressionInput', {
-  template: require('./numeric-expression-input-response.component.html'),
-  controller: ['$attrs', 'HtmlEscaperService',
-    function($attrs, HtmlEscaperService) {
-      const ctrl = this;
-      ctrl.$onInit = function() {
-        ctrl.answer = HtmlEscaperService.escapedJsonToObj($attrs.answer);
-      };
-    }
-  ]
-});
+@Component({
+  selector: 'oppia-response-numeric-expression-input',
+  templateUrl: './numeric-expression-input-response.component.html',
+  styleUrls: []
+})
+export class ResponseNumericExpressionInput implements OnInit {
+  @Input() answer: string;
+  displayAnswer: Object = '';
```

Comment:
Why is this object if it is initially set to str?

Reply:
@darksun27 where you are assigning a value to display answer (the place where you can get a type mismatch), you can use `as string` and the error will go away.

Reply:
done!